File: packages/language-go/spec/fixtures/sample.go
// Grammar tests for the Tree-sitter Go grammar.
// <- comment.line.double-slash.go

package main
// <- keyword

import (
	"fmt"
	"strings"
)

const Limit = 25
//            ^ constant.numeric.integer.go

type Counter struct {
//   ^ entity.name.type.go
	total int
	tags  []string
}

func (c *Counter) Tally(label string, step int) int {
//                ^ entity.name.function.method.go
	c.total += step
	if strings.HasPrefix(label, "x") {
		return c.total
	}
	for i := 0; i < 3; i++ {
		c.total++
	}
	fmt.Println("done", label)
	return c.total
}

func main() {
//   ^ entity.name.function.go
	c := &Counter{}
	c.Tally("a", 1)
}

File: packages/language-rust/spec/fixtures/sample.rs
// Grammar tests for the Tree-sitter Rust grammar.
// <- comment.line.double-slash.rust

/* A block comment. */
// <- comment.block.rust

use std::collections::HashMap;
// <- keyword.control

#[derive(Debug)]
//  ^ entity.other.attribute-name.rust

pub struct Counter {
//         ^ storage.type.other.rust
    total: u32,
//         ^ storage.type.builtin.rust
    tags: HashMap<String, &'static str>,
//                         ^ keyword.operator.lifetime.rust
}

impl Counter {
    pub fn tally(&mut self, label: &str, step: f64) -> u32 {
//         ^ entity.name.function.rust
//                 ^ storage.modifier.mut.rust
        let enabled = true;
//                    ^ constant.language.boolean
        let ratio = 1.5;
//                  ^ constant.numeric.decimal.float.rust
        println!("{} {}", label, step);
//      ^ support.other.function.rust
//               ^ string.quoted.double.rust
        self.total += 1;
//      ^ variable.language.self.rust
        42
//      ^ constant.numeric.decimal.integer.rust
    }
}
